File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.py
import collections
def solution(s):
    if s[0]!='(' or s[-1]!=')' or s.count('(') !=s.count(')'):
        return False
    stack=[]
    for char in s:
        if char=='(':
            stack.append('(')
        elif char==')':
            if len(stack)==0:
                return False
            stack.pop()
    return True

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.cpp
#include<string>
#include <iostream>
#include <stack>

using namespace std;

bool solution(string s)
{
    bool answer = true;
    stack<char> stack;
    
    for (int i=0;i<s.length();i++){
        if (stack.empty()){
            stack.push('(');
            continue;
        }
        
        if (s[i]=='('){
            stack.push('(');
        }else{
            if (!stack.empty() && stack.top()=='('){
                stack.pop();
            }else{
                return false;
            }
        }
    }

    return stack.empty();
}